core/trino-main/src/main/java/io/trino/execution/SqlTaskManager.java

```diff
@@ -211,7 +259,12 @@ public void start()
             catch (Throwable e) {
                 log.warn(e, "Error updating stats");
             }
-        }, 0, 1, TimeUnit.SECONDS);
+        }, 0, 1, SECONDS);
+
+        stuckSplitTaskInterrupter.ifPresent(interrupter -> {
+            long intervalSeconds = interrupter.getStuckSplitsDetectionInterval().roundTo(SECONDS);
+            taskManagementExecutor.scheduleAtFixedRate(this::failStuckSplitTasks, 0, intervalSeconds, SECONDS);
```

Comment:
let's put try-catch with a generic throwable like the above, because schedule* methods supress future executions on any sort of exception throwing